src/concrete/ml/torch/lora.py

```diff
@@ -31,49 +54,96 @@ class LoraTraining(torch.nn.Module):
     toggle between calibration and optimization modes.
 
     Args:
-        inference_model (torch.nn.Module): The base model to be fine-tuned.
-        n_layers_to_skip (int): Number of layers to skip. Linear layers that do not require
-            gradient to be propagated are skipped. Defaults to 1.
+        model (torch.nn.Module): The base model with LoRA layers to be fine-tuned.
+        n_layers_to_skip_for_backprop (int): Number of initial linear layers to keep as standard
+            layers. Since the first layer doesn't need backpropagation (no previous layer to
+            update), we typically skip 1 layer. Defaults to 1.
+        loss_fn (callable, optional): Loss function to compute the loss. If None, the model
+            is expected to return a loss.
     """
 
-    def __init__(self, inference_model, n_layers_to_skip: int = 1) -> None:
+    def __init__(self, model, n_layers_to_skip_for_backprop=1, loss_fn=None):
         super().__init__()
 
-        self.inference_model = inference_model
+        # Check if model accepts labels when no loss_fn is provided
+        if loss_fn is None:
+            from inspect import signature
```

Comment:
really not a big fan of 
```
if .... 
    from ...
```

Reply:
Yes I will change. I had to do some of those to dodge the circular import. I think this one is a mistake tho.